src/rgw/CMakeLists.txt

```diff
@@ -9,6 +9,12 @@ if(WITH_RADOSGW_SELECT_PARQUET)
   message("-- arrow is installed, radosgw/s3select-op is able to process parquet objects")
 endif(WITH_RADOSGW_SELECT_PARQUET)
 
+if(WITH_RADOSGW_ARROW_FLIGHT)
+  set(ARROW_FLIGHT_LIBRARIES utf8proc Arrow::Flight)
+  # set(ARROW_FLIGHT_LIBRARIES utf8proc::utf8proc Arrow::Flight) # this doesn't work!!
```

Comment:
I'd have thought that the commented out line would work, but linking fails with it. Linking does work with the first line, though.

Reply:
is the link error from unresolved externals, or failure to find the library?

Reply:
unresolved externals: _undefined reference to `utf8proc_toupper'_

Reply:
that is odd. you might try doing verbose builds (either `ninja -v` or `make VERBOSE=1`), and compare the two linker command lines

Reply:
So in the working version, the link command references the following in this order (with stuff before, after, in the middle): 

`src/arrow/lib/libparquet.a src/arrow /lib/libarrow_flight.a  src/arrow/lib/libarrow.a  -lutf8proc`

In the failing version we have:

`src/arrow/lib/libparquet.a  src/utf8proc/install/lib/libutf8proc.a  src/arrow/lib/libarrow_flight.a  src/arrow/lib/libarrow.a`

So it looks like an ordering issue. In fact when I changed the line to the following, it linked.
`set(ARROW_FLIGHT_LIBRARIES Arrow::Flight utf8proc::utf8proc)`

Which tells me that it's `Arrow::Flight` that should have the dependency on `utf8proc`, not `radosgw`....